src/server/dfly_main.cc

```diff
@@ -323,13 +324,30 @@ bool RunEngine(ProactorPool* pool, AcceptServer* acceptor) {
   auto port = GetFlag(FLAGS_port);
   auto mc_port = GetFlag(FLAGS_memcache_port);
   string unix_sock = GetFlag(FLAGS_unixsocket);
+  mode_t unix_socket_perm;
+  string perm_str = GetFlag(FLAGS_unixsocketperm);
+  if (!unix_sock.empty()) {
+    if (perm_str.empty()) {
+      // get umask of running process, indicates the permission bits that are turned off
+      mode_t umask_val = umask(0);
+      umask(umask_val);
+      unix_socket_perm = 0777 & ~umask_val;
+    } else {
+      if (!absl::numbers_internal::safe_strtoi_base(perm_str, &unix_socket_perm, 8) ||
+          unix_socket_perm > 0777) {
+        LOG(ERROR) << "Invalid unixsocketperm: " << perm_str;
+        exit(1);
+      }
+    }
+  }
+
   bool unlink_uds = false;
```

Comment:
You could move the definition of `unlink_uds` to line 327 and then use the same `if` (i.e. remove the `if` in line 346)